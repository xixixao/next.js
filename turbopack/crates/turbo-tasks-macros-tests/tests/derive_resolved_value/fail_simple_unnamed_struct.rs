#![allow(dead_code)]

use turbo_tasks::ResolvedValue;

struct UnresolvedValue;

#[derive(ResolvedValue)]
struct ContainsUnresolvedValueUnnamed(UnresolvedValue);

fn main() {}
