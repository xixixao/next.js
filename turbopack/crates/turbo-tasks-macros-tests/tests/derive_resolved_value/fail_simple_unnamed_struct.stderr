error[E0277]: the trait bound `UnresolvedValue: ResolvedValue` is not satisfied
 --> tests/derive_resolved_value/fail_simple_unnamed_struct.rs:8:39
  |
8 | struct ContainsUnresolvedValueUnnamed(UnresolvedValue);
  |                                       ^^^^^^^^^^^^^^^ the trait `ResolvedValue` is not implemented for `UnresolvedValue`
  |
  = help: the following other types implement trait `ResolvedValue`:
            &T
            &mut T
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
          and $N others
note: required by a bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
 --> tests/derive_resolved_value/fail_simple_unnamed_struct.rs:7:10
  |
7 | #[derive(ResolvedValue)]
  |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::assert_impl_resolved_value`
  = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)
